interop-cats/jvm/src/main/scala/scalaz/zio/interop/Schedule.scala

```diff
@@ -0,0 +1,598 @@
+
+  /**
+   * See [[scalaz.zio.ZSchedule.logInput]]
+   */
+  final def logInput[A1 <: A](f: A1 => F[Unit])(implicit R: Runtime[Env]): Schedule[F, A1, B] =
+    updated[A1, B](update => (a, s) => F.flatMap(f(a))(_ => update(a, s)))
+
+  /**
+   * See [[scalaz.zio.ZSchedule.logOutput
+   */
+  final def logOutput(f: B => F[Unit])(implicit R: Runtime[Env]): Schedule[F, A, B] =
+    updated[A, B](update => (a, s) => F.flatMap(update(a, s))(step => F.flatMap(f(step.finish()))(_ => F.pure(step))))
+
+  /**
+   * See [[scalaz.zio.ZSchedule.collect]]
+   */
+  final def collect: Schedule[F, A, List[B]] =
+    Schedule(underlying.collect)
+
+  /**
+   * See [[scalaz.zio.ZSchedule.fold]]
+   */
+  final def fold[Z](z: Z)(f: (Z, B) => Z): Schedule[F, A, Z] =
+    Schedule(underlying.fold(z)(f))
+
+  /**
+   * See [[scalaz.zio.ZSchedule.foldM]]
+   */
+  final def foldM[Z](z: F[Z])(f: (Z, B) => F[Z])(implicit R: Runtime[Env]): Schedule[F, A, Z] =
+    Schedule[F, (self.State, Z), A, Z](
+      F.flatMap(self.initial)(a => F.map(z)(b => (a, b))),
+      (a: A, s0: (self.State, Z)) => {
+        val fd: F[ZSchedule.Decision[State, B]] = self.update.apply(a, s0._1)
+        F.flatMap(fd)(step => F.map(f(s0._2, step.finish()))(z => step.bimap(s => (s, z), _ => z)))
+      }
+    )
+
+  /**
+   * See [[scalaz.zio.ZSchedule.>>>]]
+   */
+  final def >>>[C](that: Schedule[F, B, C]): Schedule[F, A, C] =
+    Schedule(underlying >>> that.underlying)
+
+  /**
+   * See [[scalaz.zio.ZSchedule.<<<]]
+   */
+  final def <<<[C](that: Schedule[F, C, A]): Schedule[F, C, B] = that >>> self
+
+  /**
+   * See [[scalaz.zio.ZSchedule.compose]]
+   */
+  final def compose[C](that: Schedule[F, C, A]): Schedule[F, C, B] = self <<< that
+
+  /**
+   * See [[scalaz.zio.ZSchedule.first]]
+   */
+  final def first[C]: Schedule[F, (A, C), (B, C)] = self *** Schedule.identity[F, C]
+
+  /**
+   * See [[scalaz.zio.ZSchedule.second]]
+   */
+  final def second[C]: Schedule[F, (C, A), (C, B)] = Schedule.identity[F, C] *** self
+
+  /**
+   * See [[scalaz.zio.ZSchedule.left]]
+   */
+  final def left[C]: Schedule[F, Either[A, C], Either[B, C]] =
+    self +++ Schedule.identity[F, C]
+
+  /**
+   * See [[scalaz.zio.ZSchedule.right]]
+   */
+  final def right[C]: Schedule[F, Either[C, A], Either[C, B]] =
+    Schedule.identity[F, C] +++ self
+
+  /**
+   * See [[scalaz.zio.ZSchedule.***]]
+   */
+  final def ***[C, D](that: Schedule[F, C, D]): Schedule[F, (A, C), (B, D)] =
+    Schedule(underlying *** that.underlying)
+
+  /**
+   * See [[scalaz.zio.ZSchedule.|||]]
+   */
+  final def |||[B1 >: B, C](that: Schedule[F, C, B1]): Schedule[F, Either[A, C], B1] =
+    (self +++ that).map(_.merge)
+
+  /**
+   * See [[scalaz.zio.ZSchedule.+++]]
+   */
+  final def +++[C, D](that: Schedule[F, C, D]): Schedule[F, Either[A, C], Either[B, D]] =
+    Schedule(underlying +++ that.underlying)
+}
+
+object Schedule {
+  import scalaz.zio.interop.catz._
+  private[interop] type Env = Random with Clock
```

Comment:
There's an open question of how to encode this for cats-effect, but I think this works for now.